test/serverTest.js

```diff
@@ -317,21 +324,28 @@ mc.supportedVersions.forEach(function (supportedVersion, i) {
           assert.strictEqual(reason, '{"text":"ServerShutdown"}')
           resolve()
         })
-        client.write('login', {
-          entityId: client.id,
+        let loginPacket = {
+          entityId: 0,
```

Comment:
Would be cleaner hers to do the same, ie dimensionCodec: loginPacket.dimensionCodec

Reply:
If I do, the code will look like that
```javascript
let loginPacket = {
  entityId: client.id,
  levelType: 'default',
  gameMode: 1,
  previousGameMode: 255,
  worldNames: ['minecraft:overworld'],
  dimensionCodec: { name: '', type: 'compound', value: { dimension: { type: 'list', value: { type: 'compound', value: [w] } } } },
  dimension: 0,
  worldName: 'minecraft:overworld',
  hashedSeed: [0, 0],
  difficulty: 2,
  maxPlayers: server.maxPlayers,
  reducedDebugInfo: 0,
  enableRespawnScreen: true
}
if (version.version >= 735) { // 1.16x
  loginPacket.dimensionCodec = mcData.loginPacket.dimensionCodec
  loginPacket.dimension = mcData.loginPacket.dimension
  loginPacket.isDebug = false
  loginPacket.isFlat = false
  loginPacket.isHardcore = false
  loginPacket.reducedDebugInfo = false
  loginPacket.viewDistance = 10
  delete loginPacket.levelType
  delete loginPacket.difficulty
}
```
It's much more verbose

Reply:
yes, verbose is not a problem. It's important to be explicit in an example
and this can be simplified a lot by doing ternary operations in each of the fields necessary

plus again `dimensionCodec: { name: '', type: 'compound', value: { dimension: { type:` this should always be taken from mcdata now